src/pages/projectSettings/tabs/EventLogTab/EventLogTab.tsx

```diff
@@ -38,11 +48,108 @@ export const EventLogTab: React.FC = () => {
     },
   });
 
+  type logEntry = {
+    timestamp: Date;
+    user: string;
+    before?: RepoEventSettingsFragment | ProjectEventSettingsFragment;
+    after?: RepoEventSettingsFragment | ProjectEventSettingsFragment;
+  };
+
+  const eventData = isRepo
+    ? repoEventData?.repoEvents?.eventLogEntries
+    : projectEventData?.projectEvents?.eventLogEntries;
+
   return (
-    <>
-      <div>This is the event log page for {identifier}</div>
-      <div> {JSON.stringify(projectEventData)}</div>
-      <div> {JSON.stringify(repoEventData)}</div>
-    </>
+    <div>
+      <div data-cy="event-log">
+        {((eventData as logEntry[]) || []).map(
+          ({ user, timestamp, before, after }) => (
+            /* @ts-expect-error */
+            <EventLogCard key={`event_log_${timestamp}`}>
+              <EventLogHeader user={user} timestamp={timestamp} />
+              <Table
+                data={getEventDiffLines(before, after)}
+                columns={[
+                  <TableHeader
+                    key="key"
+                    label={<span data-cy="key">Property</span>}
+                    sortBy={(datum: EventDiffLine) => datum?.key}
+                  />,
+                  <TableHeader
+                    key="before"
+                    label={<span data-cy="before">Before</span>}
+                    sortBy={(datum: EventDiffLine) => datum?.before}
+                  />,
+                  <TableHeader
+                    key="after"
+                    label={<span data-cy="after">After</span>}
+                    sortBy={(datum: EventDiffLine) => datum?.after}
+                  />,
+                ]}
+              >
+                {({ datum }) => (
```

Comment:
could we add a type for datum? my vsc ts compiler is having a hard time understanding the type

Reply:
datum should infer the type passed into the data prop for the table so we should type that better instead.

Reply:
It's of type EventDiffLine and my compiler picks it up. I'm not really sure where to define it better, I'm not able to add a type to that line. 

<img width="274" alt="image" src="https://user-images.githubusercontent.com/13104717/165437625-d52c8793-681c-46f8-a2de-06765fbef0a0.png">